src/server/worker/pipeline/transform/worker.go

```diff
@@ -681,24 +687,7 @@ func handleMergeTask(driver driver.Driver, logger logs.TaggedLogger, data *Merge
 				hashtreeInfo := hashtreeInfo
 				eg.Go(func() (retErr error) {
 					defer limiter.Release()
-					reader, err := fetchChunk(driver, logger, hashtreeInfo, data.Shard, data.Stats)
-					if err != nil {
-						return err
-					}
-
-					defer func() {
-						if err := reader.Close(); retErr == nil {
-							retErr = err
-						}
-					}()
-
-					// TODO: this only works if it is read into a buffer first?
-					buf := &bytes.Buffer{}
-					if _, err := io.Copy(buf, reader); err != nil {
-						return errors.EnsureStack(err)
-					}
-
-					return errors.EnsureStack(cache.Put(chunkID, bytes.NewBuffer(buf.Bytes())))
+					return errors.EnsureStack(fetchChunk(driver, logger, cache, chunkID, hashtreeInfo, data.Shard, data.Stats))
```

Comment:
I'd prefer if the `errors.EnsureStack` is closer to where the error is instantiated - if we get the stack here we won't know if it's from fetching from a worker or from a direct read without additional log statements helping out.

Reply:
We will know it is from a direct read since we will fallback on that for any error that occurs when fetching from the worker.